### PR TITLE
clean up v1 consensus half of tests are fixed

### PR DESCRIPTION
**Pull request by liam-lai**
Another PR will raise for other tests

### CODE REVIEW

File: accounts/abi/bind/backends/simulated.go [hunk 1]
```diff
@@ -109,20 +122,21 @@ func NewXDCSimulatedBackend(alloc core.GenesisAlloc, gasLimit uint64, chainConfi
...
-	lendingServ := XDCxlending.New(XDCXServ)
-
-	consensus.GetXDCXService = func() utils.TradingService {
-		return XDCXServ
-	}
-	consensus.GetLendingService = func() utils.LendingService {
-		return lendingServ
-	}
+	/*
+		// Attach mock trading and lending service
```
Comment (wjrjerome) [inline accounts/abi/bind/backends/simulated.go]: why not just delete them?

File: accounts/abi/bind/backends/simulated.go [hunk 2]
```diff
@@ -131,6 +145,14 @@ func NewXDCSimulatedBackend(alloc core.GenesisAlloc, gasLimit uint64, chainConfi
 		config:     genesis.Config,
 		events:     filters.NewEventSystem(new(event.TypeMux), &filterBackend{database, blockchain}, false),
 	}
+
+	go func() {
+		for range core.CheckpointCh {
+			checkpointChanMsg := <-core.CheckpointCh
+			log.Info("[V2] Got a message from core CheckpointChan!", "msg", checkpointChanMsg)
```
Comment (wjrjerome) [inline accounts/abi/bind/backends/simulated.go]: i'm not sure i understand the purpose of this log?

File: common/constants.go [hunk 3]
```diff
@@ -33,16 +31,16 @@ const (
...
-var TIPXDCX = big.NewInt(225000)
-var TIPXDCXLending = big.NewInt(225000)
-var TIPXDCXCancellationFee = big.NewInt(225000)
-var TIPXDCXCancellationFeeTestnet = big.NewInt(225000)
+var TIPNoHalvingMNReward = big.NewInt(0) // hardfork no halving masternodes reward
+var BlackListHFNumber = uint64(0)
+var TIPXDCX = big.NewInt(0)
+var TIPXDCXLending = big.NewInt(0)
+var TIPXDCXCancellationFee = big.NewInt(0)
+var TIPXDCXCancellationFeeTestnet = big.NewInt(0)
```
Comment (wjrjerome) [inline common/constants.go:43]: I think we should be able to remove those constants right? No need to check them in the code anymore

File: consensus/XDPoS/engines/engine_v1/engine.go [hunk 4]
```diff
@@ -78,772 +66,95 @@ panics. This is done to avoid accidentally using both forms (signature present
 or not), which could be abused to produce different hashes for the same header.
 */
 func (x *XDPoS_v1) SigHash(header *types.Header) (hash common.Hash) {
-	return sigHash(header)
+	return common.Hash{}
```
Comment (wjrjerome) [inline consensus/XDPoS/engines/engine_v1/engine.go]: Why do we still need this file? shouldn't we just remove the whole v1 directory?

File: consensus/XDPoS/engines/engine_v2/engine.go [hunk 5]
```diff
@@ -267,6 +267,10 @@ func (x *XDPoS_v2) YourTurn(chain consensus.ChainReader, parent *types.Header, s
 
 	waitedTime := time.Now().Unix() - parent.Time.Int64()
 	_, parentRound, _, err := x.getExtraFields(parent)
+	if err != nil {
+		log.Warn("[Yourturn] Error while getExtraFields", "parent", parent.Number, "error", err)
```
Comment (wjrjerome) [inline consensus/XDPoS/engines/engine_v2/engine.go:271]: `"error", err.Error()` need to convert it into string i believe

File: consensus/XDPoS/XDPoS.go [hunk 6]
```diff
@@ -208,21 +182,12 @@ func (x *XDPoS) VerifyHeaders(chain consensus.ChainReader, headers []*types.Head
 	results := make(chan error, len(headers))
 
 	// Split the headers list into v1 and v2 buckets
```
Comment (wjrjerome) [inline consensus/XDPoS/XDPoS.go]: nit: "Split the headers list into v1 and v2 buckets" can be removed

File: consensus/XDPoS/XDPoS.go [hunk 7]
```diff
@@ -346,86 +265,44 @@ func (x *XDPoS) GetMasternodesByNumber(chain consensus.ChainReader, blockNumber
...
-	switch x.config.BlockConsensusVersion(big.NewInt(parent.Number.Int64()+1), nil, SkipExtraFieldCheck) {
-	case params.ConsensusEngineVersion2:
-		return x.EngineV2.YourTurn(chain, parent, signer)
-	default: // Default "v1"
-		return x.EngineV1.YourTurn(chain, parent, signer)
-	}
+	return x.EngineV2.YourTurn(chain, parent, signer)
 }
 
 func (x *XDPoS) GetValidator(creator common.Address, chain consensus.ChainReader, header *types.Header) (common.Address, error) {
```
Comment (wjrjerome) [inline consensus/XDPoS/XDPoS.go:270]: do we still need those v1 specific methods? probably can be deleted all together right?

File: core/chain_makers.go [hunk 8]
```diff
@@ -246,6 +246,22 @@ func makeHeader(chain consensus.ChainReader, parent *types.Block, state *state.S
 		time = new(big.Int).Add(parent.Time(), big.NewInt(10)) // block time is fixed at 10 seconds
 	}
 
+	extra := types.ExtraFields_v2{
```
Comment (wjrjerome) [inline core/chain_makers.go:249]: hmm... mind if i ask why this change is needed in this PR? what's the context?

File: params/config.go
Comment (wjrjerome) [inline params/config.go:1]: `ConsensusEngineVersion1` this can be removed i believe

Comment (wjrjerome) [reply params/config.go:1]: also, for the `MainnetV2Configs`, `TestnetV2Configs` and `DevnetV2Configs` which one are we using? not necessary need to be resolved in this PR, but happy for you to create a follow up ticket to remove those configs too

File: accounts/abi/bind/backends/simulated.go (hunk 1 above)
Comment (liam-lai) [reply accounts/abi/bind/backends/simulated.go]: hmm, initially thought we will gonna use it in the future, but you are right. I can add it back whenever we need it

File: accounts/abi/bind/backends/simulated.go (hunk 2 above)
Comment (liam-lai) [reply accounts/abi/bind/backends/simulated.go]: it's copy paste from the real code, will delete it.

File: common/constants.go (hunk 3 above)
Comment (liam-lai) [reply common/constants.go:43]: It will be too much change in the current state. And lots of files will get changed. I think we can just remain this as legacy code. Also merge conflict will be less if merge from `XDPoSChain`

File: consensus/XDPoS/XDPoS.go (hunk 6 above)
Comment (liam-lai) [reply consensus/XDPoS/XDPoS.go]: 👍

File: consensus/XDPoS/XDPoS.go (hunk 7 above)
Comment (liam-lai) [reply consensus/XDPoS/XDPoS.go:270]: Just checked, to fully clean up this function `GetValidator()` will need `eth/fetcher/fetcher.go`, `eth/backend.go` changes too. I prefer keep it now. Too much changes at same time.

Comment (liam-lai) [reply consensus/XDPoS/XDPoS.go:270]: Same as other V1 functions. So just keep the function will `nil` is quick way to have light clean up

File: consensus/XDPoS/engines/engine_v2/engine.go (hunk 5 above)
Comment (liam-lai) [reply consensus/XDPoS/engines/engine_v2/engine.go:271]: In fact, golang default `error` type, if it require to be String(), then it have predefine rule to convert `err` into `err.Error()`
image

File: core/chain_makers.go (hunk 8 above)
Comment (liam-lai) [reply core/chain_makers.go:249]: because we start v2 block format at block 0 now. And some tests are using this `makeHeader` function to create a new block. And our block consensus logic require block header have V2 Extra field. So we need to change here.

File: params/config.go
Comment (liam-lai) [reply params/config.go:1]: Yes, we need to clean it. We just need mainnet config and test config. That's it. Will create jira ticket to address tihs.

File: consensus/XDPoS/engines/engine_v1/engine.go (hunk 4 above)
Comment (liam-lai) [reply consensus/XDPoS/engines/engine_v1/engine.go]: Yes, now I think the better way it just return default value at `adaptor level`

File: consensus/XDPoS/XDPoS.go (hunk 7 above)
Comment (wjrjerome) [reply consensus/XDPoS/XDPoS.go:270]: Sounds good to me. but let's track the remaining clean up in a seperate ticket. You can link the conversation from this PR in the ticket

**Review by wjrjerome — APPROVED** [review]

File: consensus/tests/engine_v2_tests/initial_test.go [hunk 9]
```diff
@@ -1,46 +1,43 @@
...
 	"github.com/stretchr/testify/assert"
 )
 
 func TestInitialFirstV2Block(t *testing.T) {
-	blockchain, _, currentBlock, _, _, _ := PrepareXDCTestBlockChainForV2Engine(t, 900, params.TestXDPoSMockChainConfig, nil)
+	blockchain, _, currentBlock, _, _, _ := PrepareXDCTestBlockChainForV2Engine(t, 1, params.TestXDPoSMockChainConfig, nil)
 	adaptor := blockchain.Engine().(*XDPoS.XDPoS)
 	header := currentBlock.Header()
-
 	// snapshot should not be created before initial
```
Comment (wgr523) [inline consensus/tests/engine_v2_tests/initial_test.go:16]: remove this comment

File: common/constants.go [hunk 10]
```diff
@@ -14,10 +14,8 @@ const (
 	EpocBlockSecret            = 800
 	EpocBlockOpening           = 850
 	EpocBlockRandomize         = 900
-	MaxMasternodes             = 18
-	MaxMasternodesV2           = 108
-	LimitPenaltyEpoch          = 4
-	LimitPenaltyEpochV2        = 0
+	MaxMasternodes             = 108
+	LimitPenaltyEpoch          = 0
```
Comment (wgr523) [inline common/constants.go:18]: `LimitPenaltyEpoch` (with or without V2) is not used anymore. We can delete it. Also, any code that references to it should be deleted.

Comment (liam-lai) [reply common/constants.go:18]: done at next pr.

File: consensus/tests/engine_v2_tests/initial_test.go (hunk 9 above)
Comment (liam-lai) [reply consensus/tests/engine_v2_tests/initial_test.go:16]: done at next pr
